source/auth/auth.rst

```diff
@@ -568,6 +568,18 @@ mechanism
 mechanism_properties
 	MUST NOT be specified.
 
+Caching credentials
+`````````````````````
+
+Drivers MUST maintain a cache of computed SCRAM credentials, where each cache
+entry's key is the tuple of ``(mongo_hashed_password, salt, i)``. Drivers SHOULD
+also include the name of the authentication mechanism, i.e. "SCRAM-SHA-1", as
```

Comment:
The name of the hash digest may be sufficient.

Reply:
I don't like that this is getting into specific mechanism like keys/tuples -- we should focus on "what", not "how".   The important thing is to be clear what factors uniquely determine a cached value.  How a library establishes uniqueness is irrelevant.

For example, the SCRAM libraries I've designed have a client object that sets password and digest on construction.  That only needs a cache based on salt and iteration values.